### PR TITLE
logging: Allow Glog reentrancy

### PR DESCRIPTION
**Pull request by theopolis**
This is an attempt to fix https://github.com/facebook/osquery/issues/3119

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit 1fee9c92a8c75b479d0477daceaad054862afe2b (Job results: [603](https://jenkins.osquery.io/job/osqueryPullRequestOSX/603/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:no_good: The commit 1fee9c92a8c75b479d0477daceaad054862afe2b (Job results: [1303](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1303/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 1fee9c92a8c75b479d0477daceaad054862afe2b (Job results: [4407](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4407/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 1fee9c92a8c75b479d0477daceaad054862afe2b (Job results: [4408](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4408/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 1fee9c92a8c75b479d0477daceaad054862afe2b (Job results: [1325](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1325/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 8489d5d7f814cf74b07aa8797b1b050d6980eaf3 (Job results: [607](https://jenkins.osquery.io/job/osqueryPullRequestOSX/607/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit b92e8377505540435d16d4f08fc6b047d128892e (Job results: [608](https://jenkins.osquery.io/job/osqueryPullRequestOSX/608/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit b92e8377505540435d16d4f08fc6b047d128892e (Job results: [4413](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4413/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit b92e8377505540435d16d4f08fc6b047d128892e (Job results: [1330](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1330/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 07428ee82a765a57286bd53bf2bd0695cf30c381 (Job results: [1331](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1331/)) failed one or more tests (**Windows**).

File: osquery/logger/logger.cpp [hunk 1]
```diff
@@ -452,14 +452,16 @@ void BufferedLogSink::WaitTillSent() {
...
+    first.wait();
+  }
 }
 
 Status LoggerPlugin::call(const PluginRequest& request,
                           PluginResponse& response) {
   if (FLAGS_logger_secondary_status_only &&
       !BufferedLogSink::isPrimaryLogger(getName()) &&
-      (request.count("string") || request.count("snapshot"))) {
+      (request.count("string") || request.count("snapsot"))) {
```
Comment (poppyseedplehzr) [inline osquery/logger/logger.cpp]: We lost an `h` here? Was this a fat-finger? Or do we really want to cont `snapsot`?

**Comment by osqueryer** [comment]
:-1: The commit 054528c82b6623199c4d4ca477eea23c085d3032 (Job results: [1332](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1332/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 64662cfc8151502ff59b9b855906613a80cb7564 (Job results: [1334](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1334/)) failed one or more tests (**Windows**).

**Pull request merged**